<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GrainGrowthWindow</class>
 <widget class="QMainWindow" name="GrainGrowthWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>819</width>
    <height>1272</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" columnstretch="1,5">
    <item row="0" column="0">
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Szerokość</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QSpinBox" name="widthInput">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>20000</number>
        </property>
        <property name="value">
         <number>300</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Wysokość</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QSpinBox" name="heightInput">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>20000</number>
        </property>
        <property name="value">
         <number>300</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QComboBox" name="startingCompositionSelect">
        <item>
         <property name="text">
          <string>Losowy z promieniem</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Losowy</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Ręczny</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Jednorodny</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QStackedWidget" name="compositionInputGroup">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="randomWithRadiusPage">
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Liczba ziaren</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QSpinBox" name="randomWithRadiusGrainCountInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20000</number>
            </property>
            <property name="value">
             <number>150</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Promień</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QSpinBox" name="radiusInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20000</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="randomPage">
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Liczba ziaren</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QSpinBox" name="randomGrainCountInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20000</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="emptyPage"/>
        <widget class="QWidget" name="homogeneousPage">
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Ziaren w poziomie</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QSpinBox" name="grainColumnsInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20000</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Ziaren w pionie</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QSpinBox" name="grainRowsInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20000</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QPushButton" name="drawButton">
        <property name="text">
         <string>Rysuj</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Warunki brzegowe</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QRadioButton" name="periodicalBcRadioButton">
           <property name="text">
            <string>Periodyczne</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QRadioButton" name="absorbingBcRadioButton">
           <property name="text">
            <string>Absorbujące</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Sąsiedztwo</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QComboBox" name="neighbourhoodSelect">
        <item>
         <property name="text">
          <string>von Neumanna</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Moore'a</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Pentagonalne</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Heksagonalne</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Promień</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QStackedWidget" name="neighbourhoodInputGroup">
        <property name="currentIndex">
         <number>4</number>
        </property>
        <widget class="QWidget" name="vonNeumannPage">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
        <widget class="QWidget" name="moorePage">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
        <widget class="QWidget" name="pentagonalPage">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QComboBox" name="pentagonalSelect">
            <item>
             <property name="text">
              <string>Górne</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Prawe</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Dolne</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Lewe</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Losowe</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="hexagonalPage">
         <layout class="QFormLayout" name="formLayout_7">
          <item row="1" column="0" colspan="2">
           <widget class="QComboBox" name="hexagonalSelect">
            <item>
             <property name="text">
              <string>Od lewej</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Od prawej</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Losowo</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="radiusPage">
         <layout class="QFormLayout" name="formLayout_8">
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Promień</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QSpinBox" name="radiusNeighbourhoodInput">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>200000</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="13" column="0" colspan="2">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Monte carlo</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Iteration count</string>
        </property>
       </widget>
      </item>
      <item row="16" column="1">
       <widget class="QSpinBox" name="monteCarloStepCountInput">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="17" column="1">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>kT</string>
        </property>
       </widget>
      </item>
      <item row="18" column="1">
       <widget class="QDoubleSpinBox" name="ktFactorInput">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>6.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="19" column="0" colspan="2">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Rekrystalizacja</string>
        </property>
       </widget>
      </item>
      <item row="20" column="1">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Iteration count</string>
        </property>
       </widget>
      </item>
      <item row="21" column="1">
       <widget class="QSpinBox" name="drxIterationCountInput">
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>200</number>
        </property>
       </widget>
      </item>
      <item row="22" column="1">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>A</string>
        </property>
       </widget>
      </item>
      <item row="23" column="1">
       <widget class="QDoubleSpinBox" name="drxAInput">
        <property name="maximum">
         <double>100000000000000000.000000000000000</double>
        </property>
        <property name="value">
         <double>86710969050178.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="24" column="1">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>B</string>
        </property>
       </widget>
      </item>
      <item row="25" column="1">
       <widget class="QDoubleSpinBox" name="drxBInput">
        <property name="decimals">
         <number>20</number>
        </property>
        <property name="maximum">
         <double>10000000000000000.000000000000000</double>
        </property>
        <property name="value">
         <double>9.412682999999999</double>
        </property>
       </widget>
      </item>
      <item row="26" column="1">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>dT</string>
        </property>
       </widget>
      </item>
      <item row="27" column="1">
       <widget class="QDoubleSpinBox" name="drxDtInput">
        <property name="decimals">
         <number>5</number>
        </property>
        <property name="value">
         <double>0.001000000000000</double>
        </property>
       </widget>
      </item>
      <item row="28" column="1">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>First dislocation pack percent size</string>
        </property>
       </widget>
      </item>
      <item row="29" column="1">
       <widget class="QDoubleSpinBox" name="drxPackPercentInput">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.700000000000000</double>
        </property>
       </widget>
      </item>
      <item row="30" column="1">
       <widget class="QLabel" name="label_19">
        <property name="text">
         <string>Random dislocation pack factor</string>
        </property>
       </widget>
      </item>
      <item row="31" column="1">
       <widget class="QDoubleSpinBox" name="drxRandomPackFactor">
        <property name="decimals">
         <number>10</number>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000010000000000</double>
        </property>
       </widget>
      </item>
      <item row="32" column="1">
       <widget class="QLabel" name="label_20">
        <property name="text">
         <string>Ro critical</string>
        </property>
       </widget>
      </item>
      <item row="33" column="1">
       <widget class="QDoubleSpinBox" name="drxRoCriticalInput">
        <property name="decimals">
         <number>10</number>
        </property>
        <property name="maximum">
         <double>10000000000000000000.000000000000000</double>
        </property>
        <property name="value">
         <double>4215840142323.419921875000000</double>
        </property>
       </widget>
      </item>
      <item row="34" column="1">
       <widget class="QLabel" name="label_21">
        <property name="text">
         <string>Grain boundary dislocation probability</string>
        </property>
       </widget>
      </item>
      <item row="35" column="1">
       <widget class="QDoubleSpinBox" name="drxBoundaryProbabilityInput">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.800000000000000</double>
        </property>
       </widget>
      </item>
      <item row="36" column="0" colspan="2">
       <widget class="QPushButton" name="simulateButton">
        <property name="text">
         <string>Rozpocznij symulację</string>
        </property>
       </widget>
      </item>
      <item row="37" column="0" colspan="2">
       <widget class="QPushButton" name="toggleViewModeButton">
        <property name="text">
         <string>Ziarna/Energia/Gęstość dyslokacji</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="QTableView" name="celluralTable">
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>57</number>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>1</number>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderMinimumSectionSize">
       <number>1</number>
      </attribute>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>819</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>widthInput</tabstop>
  <tabstop>heightInput</tabstop>
  <tabstop>startingCompositionSelect</tabstop>
  <tabstop>randomGrainCountInput</tabstop>
  <tabstop>randomWithRadiusGrainCountInput</tabstop>
  <tabstop>radiusInput</tabstop>
  <tabstop>grainColumnsInput</tabstop>
  <tabstop>grainRowsInput</tabstop>
  <tabstop>drawButton</tabstop>
  <tabstop>periodicalBcRadioButton</tabstop>
  <tabstop>absorbingBcRadioButton</tabstop>
  <tabstop>neighbourhoodSelect</tabstop>
  <tabstop>pentagonalSelect</tabstop>
  <tabstop>hexagonalSelect</tabstop>
  <tabstop>radiusNeighbourhoodInput</tabstop>
  <tabstop>monteCarloStepCountInput</tabstop>
  <tabstop>ktFactorInput</tabstop>
  <tabstop>drxIterationCountInput</tabstop>
  <tabstop>drxAInput</tabstop>
  <tabstop>drxBInput</tabstop>
  <tabstop>drxDtInput</tabstop>
  <tabstop>drxPackPercentInput</tabstop>
  <tabstop>drxRandomPackFactor</tabstop>
  <tabstop>drxRoCriticalInput</tabstop>
  <tabstop>drxBoundaryProbabilityInput</tabstop>
  <tabstop>simulateButton</tabstop>
  <tabstop>toggleViewModeButton</tabstop>
  <tabstop>celluralTable</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
